npe2/_plugin_manager.py

```diff
@@ -42,16 +60,24 @@ class PluginManager:
     _submenus: Dict[str, SubmenuContribution] = {}
     _themes: Dict[str, ThemeContribution] = {}
     _readers: DefaultDict[str, List[ReaderContribution]] = DefaultDict(list)
+    _writers_by_type: DefaultDict[
+        LayerType, IntervalTree[WriterContribution]
```

Comment:
(if you use the `IntrvlTree` def above in TYPE_CHECKING ,then change this hint here to `IntrvlTree[WriterContribution]`)